service/controller/app/v1/resource/chart/resource.go

```diff
@@ -75,16 +75,27 @@ func (r *Resource) Name() string {
 }
 
 // equals asseses the equality of ReleaseStates with regards to distinguishing fields.
-func equals(a, b *v1alpha1.Chart) bool {
-	if a.Name != b.Name {
+func equals(current, desired *v1alpha1.Chart) bool {
+	if current.Name != desired.Name {
 		return false
 	}
-	if !reflect.DeepEqual(a.Spec, b.Spec) {
+	if !reflect.DeepEqual(current.Spec, desired.Spec) {
 		return false
 	}
-	if !reflect.DeepEqual(a.Labels, b.Labels) {
+	if !reflect.DeepEqual(current.Labels, desired.Labels) {
 		return false
 	}
+
+	desiredAnnotation := desired.GetAnnotations()
+	for currentAnnotationKey, currentValue := range current.GetAnnotations() {
+		if desiredValue, ok := desiredAnnotation[currentAnnotationKey]; ok {
```

Comment:
There's a case `current` chart has some other annotations un-related to app-operator, (e.g. `chart-operator.giantswarm.io/values-md5-checksum`), we would like to skip comparing that values.